woocommerce/woocommerce-ios
# Product category: Support updating existing category

Closes: #10634 


## Description

This PR adds support for updating existing product categories. Changes include:
- Added a new swipe action on the product category list for editing categories.
- Updated the title for the delete action to show text instead of icon for clarity.
- Updated `ProductCategoryListViewModel` to support finding `ProductCategory` models based on ID.
- Renamed `AddProductCategoryViewController` to `AddEditProductCategoryViewController` to support both adding and editing categories.
- Added a new view model for `AddEditProductCategoryViewController` and updated the controller to support editing categories.
- Updated `ProductParentCategoriesViewController` to display selected item and add option to remove current parent.

## Testing instructions

- Log in to the app and navigate to the Products tab.
- Select an existing product or create a new one if needed.
- Select the category row on the product form or select Add more details > Category if the row is not already available.
- If there are no existing categories in your store, add at least one.
- On the category list, swipe left on any row. You should see a button to edit the item. Notice in Xcode console:  `🔵 Tracked product_category_settings_edit_button_tapped`.
- Tap the button, notice a new screen for editing category is displayed. The category name and parent should be correct.
- Edit any detail of the category. The Save button should only be enabled if there's any change to the category.
- Tap Save. The edit view should be dismissed and the category list should be updated.

Please feel free to check Menu > Coupon > Edit coupon > Select categories. Editing categories should not be enabled on this screen.


## Screenshots



https://github.com/woocommerce/woocommerce-ios/assets/5533851/93aba02d-a46c-4b57-b40b-c05e137cf2cd




---
